modules/system/assets/js/framework.js

```diff
@@ -41,23 +41,36 @@ if (window.jQuery.request !== undefined) {
         $triggerEl.trigger(_event, context)
         if (_event.isDefaultPrevented()) return
 
-        var data = [$form.serialize()],
-            loading = options.loading !== undefined ? options.loading : null,
+        var loading = options.loading !== undefined ? options.loading : null,
             isRedirect = options.redirect !== undefined && options.redirect.length,
-            useFlash = options.flash !== undefined
+            useFlash = options.flash !== undefined,
+            sendFiles = $el.is(':file') || ($el[0] == $form[0] && $form.attr('enctype') == 'multipart/form-data') || options.sendFiles,
+            data = $triggerEl.find(':input').addBack(':input').serializeArray()
```

Comment:
Think I've tried to be too clever here. If using `$.request` it will find all the elements in the document and submit them, pretty sure this isn't desired behaviour.